engine/src/mac-core.mm

```diff
@@ -799,8 +801,21 @@ bool MCPlatformWaitForEvent(double p_duration, bool p_blocking)
 		// Run the modal session, if it has been created yet (it might not if this
 		// wait was triggered by reacting to an event caused as part of creating
 		// the modal session, e.g. when losing window focus).
-		if (s_modal_sessions[s_modal_session_count - 1].session != nil)
-			[NSApp runModalSession: s_modal_sessions[s_modal_session_count - 1] . session];
+		if (s_modal_session_run_depth < s_modal_session_count && s_modal_sessions[s_modal_session_run_depth].session != nil)
```

Comment:
Can you add to the comment here to explain what the modal session run_depth guard is doing.